test-framework/vault/src/main/java/io/quarkus/vault/test/VaultTestExtension.java

```diff
@@ -495,6 +526,9 @@ public void close() {
         if (postgresContainer != null && postgresContainer.isRunning()) {
             postgresContainer.stop();
         }
+        if (rabbitMQContainer != null && rabbitMQContainer.isRunning()) {
+            rabbitMQContainer.stop();
+        }
```

Comment:
I'm a bit puzzled by this class. Basically it, unconditionally, starts a bunch of containers. how would a regular Vault user use this class?

Reply:
This is only to support `VaultITCase`. It starts a lot of containers because it's testing Vault which touches a lot of extensions via the credentials provider mechanism. 

A "normal user" would use this by booting their Vault & RabbitMQ instances manually (or k8s, Terraform, etc.) then configure the RabbitMQ secrets engine to target their RabbitMQ instance.

This is no different than the way database dynamic credentials works; see the lines above that boot/shutdown PostgreSQL. User's using Vault for dynamic credentials are very accustomed to these configuration steps as they are required.

As an aside, we don't currently ever use dynamic credentials during local & CI testing explicitly because it is cumbersome to setup. Dynamic credentials usage starts only after deploy where the Vault, RabbitMQ, PostgreSQL and others already exist and Vault is already configured. That's why contributing this to Quarkus where it can be tested properly was important to us.

Reply:
I would like @vsevel 's take on this one. @vsevel is `VaultTestExtension` only there to support internal Vault testing? Or it's supposed to be consumable by others for some testing as well? If so, I'm wondering if we really want to start all these containers always.

Reply:
@gsmet Currently `VaultTestExtension` is used in the `vault`, & `vault-agroal` integration tests. So I would say it's for internal testing but needed to be shared between multiple test modules.

There is an argument to be made that this is adding "another" container to all the Vault tests but since Quarkus/TestContainers only start them once before all the tests are run, I can't see that it's adding a real resource drain because RabbitMQ would need to be started and stopped at least for it's test.

Reply:
@gsmet I confirm `VaultTestExtension` has been designed for internal testing only.
I did not have a look at the PR so far. I was waiting for tests to pass, and let me know whenever you were ready @kdubb 
Please advise. thanks.

Reply:
@vsevel I think it's ready. The only failing test was an unrelated timeout in the JDK11 Windows build.

Reply:
OK, so my issue with this work is that if we are moving Vault out (and the plan is to do it), we can't rely on Vault artifacts in the Core. These tests would have to be moved out of the Core and inside the Vault extension.

While I agree it's handy to use Vault for testing this. I think you should have a very basic credential provider implemented in your tests instead, one that could be entirely static and not depending on Vault.

That is best to properly modularize things.

Does it make sense?

Reply:
Also, we should test in Core with a dummy credentials provider but I have nothing against you moving the more advanced using Vault to the Vault Quarkiverse extension.

Reply:
@gsmet I've removed all Vault testing and replaced it with a test credentials provider bean added in the test package. It's enough to prove the credentials provider is working without requiring any external dependencies.